File: app/src/androidTest/java/com/outplaysoftworks/sidedeck/LpLogTests.java
package com.outplaysoftworks.sidedeck;

import android.annotation.SuppressLint;
import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;
import android.support.test.rule.ActivityTestRule;
import android.support.test.runner.AndroidJUnit4;

import org.junit.After;
import org.junit.Before;
import org.junit.Rule;
import org.junit.runner.RunWith;

import static android.support.test.InstrumentationRegistry.getInstrumentation;

@RunWith(AndroidJUnit4.class)
//@LargeTest
public class LpLogTests {
    private int durationSleepPadding = 250;

    @Rule
    public ActivityTestRule<MainActivity> mActivityRule = new ActivityTestRule<>(
            MainActivity.class);


    @SuppressLint("ApplySharedPref")
    @Before
    public void initSettingsForTesting() {
        Context context = getInstrumentation().getTargetContext();
        SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.putBoolean(context.getString(R.string.KEYallowNegativeLp), false);
        editor.putString(context.getString(R.string.KEYplayerOneDefaultNameSetting), "Player 1");
        editor.putString(context.getString(R.string.KEYplayerTwoDefaultNameSetting), "Player 2");
        editor.putString(context.getString(R.string.KEYdefaultLpSetting), "8000");
        editor.commit();
    }

    @After
    public void cleanUpModel() {
        LpCalculatorModel.clearEnteredValue();
        LpCalculatorModel.setPlayer1Name(LpCalculatorModel.getPlayer1Name());
        LpCalculatorModel.setPlayer1Lp(LpCalculatorModel.getPlayer1Lp());
        LpCalculatorModel.setPlayer2Lp(LpCalculatorModel.getPlayer2Lp());
        LpCalculatorModel.setAllowsNegativeLp(false);
        LpCalculatorModel.setLpDefault(8000);
    }

}

File: app/src/androidTest/java/com/outplaysoftworks/sidedeck/LpLogTests.kt
package com.outplaysoftworks.sidedeck

import android.annotation.SuppressLint
import android.content.Context
import android.content.SharedPreferences
import android.preference.PreferenceManager
import android.support.test.rule.ActivityTestRule
import android.support.test.runner.AndroidJUnit4

import org.junit.After
import org.junit.Before
import org.junit.Rule
import org.junit.runner.RunWith

import android.support.test.InstrumentationRegistry.getInstrumentation

@RunWith(AndroidJUnit4::class)
//@LargeTest
class LpLogTests {
    private val durationSleepPadding = 250

    @Rule
    var mActivityRule = ActivityTestRule<MainActivity>(
            MainActivity::class.java)


    @SuppressLint("ApplySharedPref")
    @Before
    fun initSettingsForTesting() {
        val context = getInstrumentation().targetContext
        val sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context)
        val editor = sharedPreferences.edit()
        editor.putBoolean(context.getString(R.string.KEYallowNegativeLp), false)
        editor.putString(context.getString(R.string.KEYplayerOneDefaultNameSetting), "Player 1")
        editor.putString(context.getString(R.string.KEYplayerTwoDefaultNameSetting), "Player 2")
        editor.putString(context.getString(R.string.KEYdefaultLpSetting), "8000")
        editor.commit()
    }

    @After
    fun cleanUpModel() {
        LpCalculatorModel.clearEnteredValue()
        LpCalculatorModel.player1Name = LpCalculatorModel.player1Name
        LpCalculatorModel.player1Lp = LpCalculatorModel.player1Lp
        LpCalculatorModel.player2Lp = LpCalculatorModel.player2Lp
        LpCalculatorModel.allowsNegativeLp = false
        LpCalculatorModel.lpDefault = 8000
    }

}